library/systemd/src/lib/yast2/systemd/service.rb

```diff
@@ -0,0 +1,216 @@
+    #    # To get an overview of available service properties, try e.g., `systemctl show sshd.service`
+    #
+    #    service = Yast::Systemd::Service.find('sshd', :type=>'Type')
+    #    service.properties.type  # 'simple'
+    class Service < Unit
+      Yast.import "Stage"
+      include Yast::Logger
+
+      UNIT_SUFFIX = ".service".freeze
+
+      class << self
+        # @param service_name [String] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Service,nil] `nil` if not found
+        def find(service_name, propmap = {})
+          service = build(service_name, propmap)
+          return nil if service.properties.not_found?
+          service
+        end
+
+        # @param service_name [String] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Service]
+        # @raise [Systemd::ServiceNotFound]
+        def find!(service_name, propmap = {})
+          find(service_name, propmap) || raise(Systemd::ServiceNotFound, service_name)
+        end
+
+        # @param service_names [Array<String>] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Array<Service,nil>] `nil` if a service is not found,
+        #   [] if this helper cannot be used:
+        #   either we're in the inst-sys without systemctl,
+        #   or it has returned fewer services than requested
+        #   (and we cannot match them up)
+        def find_many_at_once(service_names, propmap = {})
+          return [] if Yast::Stage.initial
+
+          snames = service_names.map { |n| n + UNIT_SUFFIX unless n.end_with?(UNIT_SUFFIX) }
+          snames_s = snames.join(" ")
+          pnames_s = Systemd::Unit::DEFAULT_PROPMAP.merge(propmap).values.join(",")
+          out = Systemctl.execute("show  --property=#{pnames_s} #{snames_s}")
+          log.error "returned #{out.exit}, #{out.stderr}" unless out.exit.zero? && out.stderr.empty?
+          property_texts = out.stdout.split("\n\n")
+          return [] unless snames.size == property_texts.size
+
+          snames.zip(property_texts).each_with_object([]) do |(name, property_text), memo|
+            service = new(name, propmap, property_text)
+            memo << service unless service.not_found?
+          end
+        end
+
+        # @param service_names [Array<String>] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Array<Service,nil>] `nil` if not found
+        def find_many(service_names, propmap = {})
+          services = find_many_at_once(service_names, propmap)
+          return services unless services.empty?
+
+          log.info "Retrying one by one"
+          service_names.map { |n| find(n, propmap) }
+        end
+
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Array<Service>]
+        def all(propmap = {})
+          Systemctl.service_units.map { |service_unit| new(service_unit, propmap) }
```

Comment:
Please, follow the [SUSE Ruby style guide](https://github.com/SUSE/style-guides/blob/master/Ruby.md) and use a single letter identifier (see the [parameters](https://github.com/SUSE/style-guides/blob/master/Ruby.md#parameters) section).